engine/access/rest/websockets/data_providers/args_validation.go

```diff
@@ -1,57 +1,191 @@
 package data_providers
 
 import (
+	"errors"
 	"fmt"
 	"strconv"
 
 	"github.com/onflow/flow-go/engine/access/rest/common"
+	httpmodels "github.com/onflow/flow-go/engine/access/rest/http/models"
 	"github.com/onflow/flow-go/engine/access/rest/websockets/models"
+	"github.com/onflow/flow-go/model/flow"
 )
 
+// ErrInvalidArgument is a sentinel error returned by argument validation helpers in this package.
+//
+// Expected errors:
+//   - [data_providers.ErrInvalidArgument]: A client supplied an invalid or unsupported argument.
+//
+// All validation errors produced by functions like ensureAllowedFields, extractArrayOfStrings,
+// extractHeartbeatInterval, and extractExecutionStateQueryFields must satisfy
+// errors.Is(err, ErrInvalidArgument). This allows callers and tests to reliably classify
+// client-input errors without relying on exact error message matching.
```

Comment:
I think this section is obvious for an experienced go developer, and doesn't need to be explicitly stated.
```suggestion
```